src/gl_context.rs

```diff
@@ -61,6 +65,7 @@ impl<Native> GLContext<Native>
         let mut context = try!(Self::create(shared_with));
 
         context.formats = GLFormats::detect(&attributes);
+        context.limits = GLLimits::detect();
```

Comment:
There should be no need to re-detect the limits here.

The formats are needed because the new attributes don't have to be the old ones, but the limits should always be the same.